extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -118,7 +118,8 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
                     throw new RuntimeException(e);
                 }
             }
-            Handler<RoutingContext> handler = recorder.handler(cached.cachedDirectory, swaggerUiConfig.path);
+            Handler<RoutingContext> handler = recorder.handler(cached.cachedDirectory,
+                    httpRootPathBuildItem.adjustPath(swaggerUiConfig.path));
             routes.produce(new RouteBuildItem(swaggerUiConfig.path, handler));
             routes.produce(new RouteBuildItem(swaggerUiConfig.path + "/*", handler));
             displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(swaggerUiConfig.path + "/"));
```

Comment:
I think at least this one needs it.

Reply:
This gets adjusted in the processor that handles the routes, but turns out none of the others do.

Reply:
I am going to open a new PR for the fixing the not found page, as it is not really related.